server/src/main/java/org/elasticsearch/index/IndexSettings.java

```diff
@@ -653,6 +655,55 @@ public Iterator<Setting<?>> settings() {
         Property.Final
     );
 
+    public static final Setting<Boolean> RECOVERY_USE_SYNTHETIC_SOURCE_SETTING = Setting.boolSetting(
+        "index.recovery.use_synthetic_source",
```

Comment:
I think if the leader index uses synthetic source for recovery then the follower index should use that too?

Reply:
I don't have a strong opinion but wanted to mention it to be sure that it is a deliberated choice.

Reply:
Are we replicating the `INDEX_MAPPER_SOURCE_MODE_SETTING`? It's mandatory for this to work since this new setting cannot be enabled if the source is not synthetic.

Reply:
INDEX_MAPPER_SOURCE_MODE_SETTING is replicated.

Reply:
Oh, I just discovered that we enforce specific checks for INDEX_MAPPER_SOURCE_MODE_SETTING to avoid changes during restores (#115811) and resizes (#115812).

I suppose we need similar check for the new setting, WDYT @martijnvg ?

Reply:
Good catch, I pushed https://github.com/elastic/elasticsearch/pull/114618/commits/fdfb2cb32b693a4ab64db14985a14421ef8b80ea

Reply:
good catch @tlrx!